.github/workflows/build.yml

```diff
@@ -401,9 +410,10 @@ jobs:
 
   integration-tests-success:
     needs:
+      - conditional-skip
       - integration-tests
     runs-on: ubuntu-latest
-    if: always()
+    if: always() && needs.conditional-skip.outputs.skip-ci != 'true'
```

Comment:
So the action won't get stuck because this job is required and there are cases where we won't run it, correct?

Reply:
Exactly. This follows the pattern in `consul` [here](https://github.com/hashicorp/consul/blob/7a19d2e7a48ffc7a6b16889adad63b121e98d82c/.github/workflows/test-integrations.yml#L599)